src/main/java/bio/terra/service/snapshot/flight/delete/SnapshotDeleteFlight.java

```diff
@@ -103,6 +103,10 @@ public SnapshotDeleteFlight(FlightMap inputParameters, Object applicationContext
     // deletes the snapshot group, so no ACL cleanup is needed beyond that.
     addStep(new DeleteSnapshotAuthzResource(iamClient, snapshotId, userReq));
 
+    // Now that we no longer have the resources in SAM, we can delete the underlying Sam group
+    // that was created for snapshots byRequestId
+    addStep(new DeleteSnapshotDeleteSamGroupStep(iamClient, snapshotId));
```

Comment:
do we want to specify here that this only affects snapshots created by the ByRequestId mode

Reply:
The javadoc comment on the constructor of this step specifies that it is only for byRequestId. You can see this comment when you over the step in the Flight in intellij.